test-support/test_oidc_provider.js

```diff
@@ -7,30 +7,32 @@ const configuration = {
       response_types: ['id_token'],
       response_mode: ['form_post'],
       grant_types: ['implicit'],
-      // "native" because we're on localhost.
-      application_type: 'native',
+      application_type: 'web',
       scopes: ['openid', 'profile'],
       redirect_uris: [
         'http://localhost:9000/callback/OidcClient',
         'http://localhost:9000/callback/AdClient',
         'http://localhost:19001/callback/OidcClient',
         'http://localhost:19001/callback/AdClient',
+        'http://civiform:9000/callback/OidcClient',
+        'http://civiform:9000/callback/AdClient',
       ],
     },
   ],
 
   // Required method, we fake the account details.
   async findAccount(ctx, id) {
+    const email = `${id}@example.com`
     return {
       accountId: id,
       async claims() {
         return {
           sub: id,
           // pretend to be IDCS which uses this key for user email.
-          user_emailid: id + '@example.com',
+          user_emailid: email,
           // lie about verification for tests.
           email_verified: true,
-          user_displayname: 'first middle last',
+          user_displayname: email,
```

Comment:
can you make this a string with more than 1 part? "user $email", etc.

  It'll help visual testing since names will often be multipart.

Reply:
This one's actually kind of fun. For AWS staging which uses Auth0, the returned "name" attribute will always be ONLY the email address (`foo@example.com`). For Seattle staging, it's a `first middle last` setup. We have server-side code that handles each of these cases [here](https://github.com/civiform/civiform/blob/5bc4314e36da0b4b2d02f60c2bfc8fdeb8ab94d9/server/app/services/applicant/ApplicantData.java#L87-L109).

Given that values here are pretty flexible, I'd personally prefer to go for the "simpler" approach of a single part, since that's an equally valid scenario.

I left a comment as such. If you feel strongly, let me know and I can try and switch to multi-part. Given that we only have a single ID and we want to be able to differentiate between different logged-in guests, I'm not sure what we'd want to use for the second part.

Reply:
Progress is better, so ok, but for manual testing having multiple name parts will allow for dev testing flow issues that might break up the content or wrap it oddly.